scripts/emulate_network.go

```diff
@@ -0,0 +1,233 @@
+						}
+					}
+					extraParams = append(extraParams, imageArgs)
+					p["el_extra_params"] = extraParams
+				} else {
+					fmt.Println("Error: el_extra_params is not an array of strings")
+				}
+				break // Exit the loop once the participant is found and updated
+			}
+		}
+	} else {
+		fmt.Println("Participants field not found or not an array")
+		return
+	}
+
+	// Marshal the map back to JSON
+	modifiedContent, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		fmt.Println("Error marshalling the map back to JSON:", err)
+		return
+	}
+
+	// Print out the resulting string
+	fmt.Println(string(modifiedContent))
+
+	// Save the modified content back to the file
+	//err = ioutil.WriteFile("network_params.json", modifiedContent, os.ModePerm)
+	if err != nil {
+		fmt.Println("Error writing the modified content to the file:", err)
+	}
+}
+
+func run(imageTag, imageArgs, enclaveName string, maxSteps int, kurtosisPath, kurtosisNetworkScriptFolder string, kurtosisNetConfigPath string) {
+	/*	params := map[string]interface{}{
+		"image_tag":               imageTag,
+		"image_args":              imageArgs,
+		"enclave_name":            enclaveName,
+		"max_steps":               maxSteps,
+		"kurtosis_path":           kurtosisPath,
+		"kurtosis_network_config": kurtosisNetworkScriptFolder,
+	}*/
+	update_config(imageTag, imageArgs, kurtosisNetConfigPath)
+
+	cmd := fmt.Sprintf("%s run --enclave %s %s", kurtosisPath, enclaveName, kurtosisNetworkScriptFolder)
+	runCommand(cmd)
+}
+
+func stop(kurtosisPath, enclaveName string) {
+	if enclaveName == "" {
+		fmt.Println("Error: enclave name must be specified.")
+		cmd := fmt.Sprintf("%s enclave ls", kurtosisPath)
+		runCommand(cmd)
+		return
+	}
+	cmd := fmt.Sprintf("%s enclave rm -f %s", kurtosisPath, enclaveName)
+	runCommand(cmd)
+}
+
+func runCommand(cmd string) {
+	var command *exec.Cmd
+	if runtime.GOOS == "windows" {
+		fmt.Println("Running on Windows:", cmd)
+		command = exec.Command("cmd", "/C", cmd)
+	} else {
+		fmt.Println("Running:", cmd)
+		command = exec.Command("sh", "-c", cmd)
+	}
+
+	stdoutPipe, err := command.StdoutPipe()
+	if err != nil {
+		fmt.Printf("Error obtaining stdout: %v\n", err)
+		return
+	}
+
+	stderrPipe, err := command.StderrPipe()
+	if err != nil {
+		fmt.Printf("Error obtaining stderr: %v\n", err)
+		return
+	}
+
+	err = command.Start()
+	if err != nil {
+		fmt.Printf("Command start failed: %v\n", err)
+		return
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		scanner := bufio.NewScanner(stdoutPipe)
+		for scanner.Scan() {
+			fmt.Println(scanner.Text())
+		}
+	}()
+
+	go func() {
+		defer wg.Done()
+		scanner := bufio.NewScanner(stderrPipe)
+		for scanner.Scan() {
+			fmt.Println(scanner.Text())
+		}
+	}()
+
+	wg.Wait()
+
+	err = command.Wait()
+	if err != nil {
+		fmt.Printf("Command execution failed: %v\n", err)
+	}
+}
+
+func help() {
+	fmt.Println(`Emulate Network script
+Available commands:
+- build
```

Comment:
This will lead to undesired and unexpected behavior scenarios:
```
Q: why did my start fail? - it worked 5 minutes ago!
A: you changed a single file in the builder codebase.
Q I did not tell you to do anything related to my docker tag XXX!
```